File: CarRentalAPI/CarRentalAPI/Data/Model/Reservation.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace CarRentalAPI.Data.Model
{
    public class Reservation
    {
        [Key]
        public int Id { get; set; }
        public string PickUpLocation { get; set; }
        public string ReturnLocation { get; set; }
        public DateTime PickUpDate { get; set; }
        public DateTime ReturnDate { get; set; }

        public int ClientId { get; set; }
        public Client Client { get; set; }

        public int CarId { get; set; }
        public Car Car { get; set; }
    }
}

File: CarRental/src/app/models/reservation.ts
import { Car } from "./car";

export class Reservation {
  id: number;
  pickUpLocation: string;
  returnLocation: string;
  pickUpDate: Date;
  returnDate: Date;
  clientLastName: string;
  clientAge: number;
  carId: number;
  car: Car;
}
